pytorch_lightning/overrides/base.py

```diff
@@ -19,25 +21,61 @@
 from pytorch_lightning.utilities.device_dtype_mixin import DeviceDtypeModuleMixin
 
 
-class _LightningModuleWrapperBase(DeviceDtypeModuleMixin, torch.nn.Module):
+class _LightningPrecisionModuleWrapperBase(torch.nn.Module):
```

Comment:
Also maybe we should also derive this from `DeviceDtypeMixin` since I am not sure, all the device and dtype changes are forwarded correctly (I think per default pytorch only traverses tensors (including children tensors) but not the modules themselves, meaning that the properties would not be updated. Mind also adding a test for this?

Reply:
Good catch! I've added some tests :smiley: